app/controllers/foreman_tasks/concerns/hosts_controller_extension.rb

```diff
@@ -8,16 +8,20 @@ module HostsControllerExtension
       end
 
       def facts_with_dynflow
-        task = ForemanTasks.async_task(::Actions::Foreman::Host::ImportFacts,
-                                       detect_host_type,
-                                       params[:name],
-                                       params[:facts],
-                                       params[:certname],
-                                       detected_proxy.try(:id))
+        type = params.fetch(:facts, {}).fetch(:_type, {}) || 'puppet'
+        trigger_type = FactImporter.importer_for(type).support_background ? :async_task : :sync_task
+
+        task = ForemanTasks.public_send(trigger_type,
```

Comment:
+1 for that, but should I push it into this PR? would that deserve some version bump? it seems like public API